src/services/createRouterRoute.ts

```diff
@@ -39,16 +42,16 @@ export type RouterRoute<TRoute extends ResolvedRoute = ResolvedRoute> = Readonly
   update: RouteUpdate<TRoute>,
 
   /**
-   * A private symbol for type guarding withing this project
+   * A private property for type guarding withing this project
    */
   [isRouterRouteSymbol]: true,
-}>
+}
 
 export function isRouterRoute(value: unknown): value is RouterRoute {
   return typeof value === 'object' && value !== null && isRouterRouteSymbol in value && value[isRouterRouteSymbol] === true
 }
 
-export function createRouterRoute<TRoute extends ResolvedRoute>(route: TRoute, push: RouterPush): RouterRoute<TRoute> {
+export function createRouterRoute<TRoute extends ResolvedRoute>(route: TRoute, push: RouterPush<Route<string, Path, Query, false>[]>): RouterRoute<TRoute> {
```

Comment:
you'll see this in a couple places, the default `Route` (and `Routes`) assumes key might be undefined and disabled might be true. This version is more accurate of a "resolved route" where the route must be named and not disabled